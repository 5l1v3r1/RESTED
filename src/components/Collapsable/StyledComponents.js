import styled from 'styled-components';

/* eslint-disable import/prefer-default-export */
export const StyledCollapsable = styled.div`
  h4 {
    margin: 0;
    color: black;
  }
  i {
    margin-left: 5px;
    transition: .25s ease-in-out;
  }
`;

