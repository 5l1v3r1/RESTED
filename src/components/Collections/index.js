import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import * as Actions from 'store/collections/actions';
import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix pull-right">
      <button onClick={addCollection}>
        Add new collection
      </button>
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel header={<Titlebar addCollection={addCollection} />}>
      <CollectionList />
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

