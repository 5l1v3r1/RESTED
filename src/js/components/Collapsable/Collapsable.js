import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Button, Collapse, Row, Col } from 'react-bootstrap';

import * as Actions from '../../store/collapsable/actions';

export function Collapsable({ id, title, open, children, toggleCollapse }) {
  return (
    <Row>
      <Button
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, !!open);
        }}
      >
        <h4>
          {title}
          <i
            className={classNames('fa fa-angle-right', {
              'fa-rotate-90': open,
            })}
          />
        </h4>
      </Button>
      <Collapse in={open}>
        <Col xs={12}>
          {children}
        </Col>
      </Collapse>
    </Row>
  );
}

Collapsable.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  open: PropTypes.bool,
};

const mapStateToProps = ({ collapsable }, { id }) => ({
  open: collapsable && collapsable[id] && collapsable[id].expanded,
});

const mapDispatchToProps = dispatch => ({
  toggleCollapse(id, open) {
    const action = open
      ? Actions.collapse(id)
      : Actions.expand(id);
    dispatch(action);
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Collapsable);

