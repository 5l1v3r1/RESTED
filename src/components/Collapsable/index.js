import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Button, Collapse, Row, Col } from 'react-bootstrap';

import Fonticon from 'components/Fonticon';
import * as Actions from 'store/config/actions';
import { isOpen } from 'store/config/selectors';

import { StyledCollapsable } from './StyledComponents';

export function Collapsable({ id, title, open, children, toggleCollapse }) {
  return (
    <StyledCollapsable>
      <Button
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, open);
        }}
      >
        <h4>
          {title}
          <Fonticon
            icon="angle-right"
            className={classNames({
              'fa-rotate-90': open,
            })}
          />
        </h4>
      </Button>
      <Collapse in={open}>
        <Row>
          <Col xs={12}>
            {children}
          </Col>
        </Row>
      </Collapse>
    </StyledCollapsable>
  );
}

Collapsable.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  open: PropTypes.bool,
};

const mapStateToProps = (state, props) => ({
  open: isOpen(state, props),
});

export default connect(mapStateToProps, Actions)(Collapsable);

