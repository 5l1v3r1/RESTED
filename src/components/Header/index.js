import React from 'react';
import { StyledHeader } from './StyledComponents';

export default function Header() {
  return (
    <StyledHeader>
      <h1>
        <img
          className="logo"
          role="presentation"
          height="40"
          src="img/rested-logo.png"
        />
        <span>RESTED</span>
      </h1>
    </StyledHeader>
  );
}

