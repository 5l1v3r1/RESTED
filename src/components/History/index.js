import React from 'react';

import { Panel } from './StyledComponents';
import HistoryList from './HistoryList';

export default function History() {
  return (
    <Panel header="">
      <HistoryList />
    </Panel>
  );
}

