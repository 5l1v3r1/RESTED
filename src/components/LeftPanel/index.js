import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { NavItem } from 'react-bootstrap';

import Collections from 'components/Collections';
import History from 'components/History';
import { updateOption } from 'store/options/actions';
import { getActiveTab } from 'store/options/selectors';

import { Panel, StyledTitlebar } from './StyledComponents';

function Titlebar(props) {
  return (
    <StyledTitlebar {...props} bsStyle="pills">
      <NavItem eventKey="collections">Collections</NavItem>
      <NavItem eventKey="history">History</NavItem>
    </StyledTitlebar>
  );
}

function LeftPanel({ activeTab, setActiveTab }) {
  return (
    <Panel header={<Titlebar activeKey={activeTab} onSelect={setActiveTab} />}>
      {activeTab === 'collections' && <Collections />}
      {activeTab === 'history' && <History />}
    </Panel>
  );
}

LeftPanel.propTypes = {
  activeTab: PropTypes.string.isRequired,
  setActiveTab: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  activeTab: getActiveTab(state),
});

export default connect(mapStateToProps, {
  setActiveTab: value => updateOption('activeTab', value),
})(LeftPanel);

