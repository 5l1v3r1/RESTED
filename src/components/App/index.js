import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Row } from 'react-bootstrap';
import HTML5Backend from 'react-dnd-html5-backend';
import { DragDropContext } from 'react-dnd';
import flow from 'lodash.flow';

import Header from 'components/Header';
import LeftPanel from 'components/LeftPanel';
import Request from 'components/Request';
import Response from 'components/Response';
import Modal from 'components/Modal';
import updateTheme from 'utils/updateTheme';
import { getTheme, getCollectionsMinimized } from 'store/options/selectors';
import { fetchOptions } from 'store/options/actions';
import { fetchUrlVariables } from 'store/urlVariables/actions';
import { THEMES } from 'constants/constants';
import './GlobalStyles';

import { LeftCol, RightCol } from './StyledComponents';

/*
 * This must be a React.Component because DragDropContext
 * attaches a ref to the component, which as we know will
 * not work with a stateless functional component.
 */
class App extends React.Component {
  constructor(props) {
    super(props);
    props.fetchOptions();
    props.fetchUrlVariables();

    updateTheme(props.theme);
  }

  componentWillReceiveProps(newProps) {
    if (newProps.theme !== this.props.theme) {
      updateTheme(newProps.theme);
    }
  }

  render() {
    const { collectionsMinimized } = this.props;

    return (
      <div>
        <Header />
        <Row>
          <LeftCol
            xsHidden
            sm={collectionsMinimized ? null : 4}
            collapsed={collectionsMinimized}
          >
            <aside>
              <LeftPanel />
            </aside>
          </LeftCol>
          <RightCol
            xs={12}
            sm={collectionsMinimized ? 12 : 8}
          >
            <main>
              <Request />
              <Response />
            </main>
          </RightCol>
        </Row>
        <Modal />
      </div>
    );
  }
}

App.propTypes = {
  fetchUrlVariables: PropTypes.func.isRequired,
  fetchOptions: PropTypes.func.isRequired,
  theme: PropTypes.oneOf(THEMES).isRequired,
  collectionsMinimized: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  theme: getTheme(state),
  collectionsMinimized: getCollectionsMinimized(state),
});

export default flow(
  connect(mapStateToProps, { fetchOptions, fetchUrlVariables }),
  DragDropContext(HTML5Backend),
)(App);

