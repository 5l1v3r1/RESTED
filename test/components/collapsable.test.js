import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import { Collapsable } from 'components/Collapsable';

it('should render correctly', () => {
  const tree = renderer.create(
    <Collapsable
      id="test"
      title="Test"
      toggleCollapse={() => {}}
    >
      <h3>test</h3>
    </Collapsable>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call the toggleCollapse function when activated', () => {
  const toggleCollapse = jest.fn();

  let wrapper = mount(
    <Collapsable
      id="test"
      title="Test"
      toggleCollapse={toggleCollapse}
    >
      <h3>test</h3>
    </Collapsable>
  );

  expect(toggleCollapse).not.toHaveBeenCalled();

  wrapper.find('button').simulate('click');
  expect(toggleCollapse).toHaveBeenCalled();
  expect(toggleCollapse).toHaveBeenCalledWith('test', false);

  toggleCollapse.mockClear();
  wrapper = mount(
    <Collapsable open
      id="test2"
      title="Test"
      toggleCollapse={toggleCollapse}
    >
      <h3>test</h3>
    </Collapsable>
  );

  expect(toggleCollapse).not.toHaveBeenCalled();

  wrapper.find('button').simulate('click');
  expect(toggleCollapse).toHaveBeenCalled();
  expect(toggleCollapse).toHaveBeenCalledWith('test2', true);
});

