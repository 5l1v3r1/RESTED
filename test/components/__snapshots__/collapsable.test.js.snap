// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="NupZf"
>
  <button
    className="btn btn-link"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <h4>
      Test
      <i
        className="fa fa-angle-right"
        role="presentation"
      />
    </h4>
  </button>
  <div
    aria-expanded={null}
    className="collapse col-xs-12"
  >
    <h3>
      test
    </h3>
  </div>
</div>
`;
